### PR TITLE
make timezone field searchable

### PR DESCRIPTION
**Pull request by sravfeyn**
http://manage.dimagi.com/default.asp?147650#919816

### CODE REVIEW

**Comment by esoergel** [comment]
Haha awesome.  Love it when libraries Just Work.

File: corehq/apps/registration/templates/registration/domain_request.html [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
 
 {% block title %}New Project Space{% endblock title %}
 
+{% block js %}{{ block.super }}
+    <script src="{% static 'hqwebapp/js/lib/select2/select2.js' %}"></script>
+{% endblock %}
+
+{% block head %}{{ block.super }}
+    <link href="{% static 'hqwebapp/js/lib/select2/select2.css' %}" rel="stylesheet" />
+{% endblock %}
```
Comment (esoergel) [inline corehq/apps/registration/templates/registration/domain_request.html]: are these two dependencies not included in our root template?

Comment (sravfeyn) [inline corehq/apps/registration/templates/registration/domain_request.html]: I think not (try `$("#id_domain_timezone").select2()` on console on [register-domain page](https://www.commcarehq.org/register/domain/commcare)), isn't that how it should be?

Comment (esoergel) [inline corehq/apps/registration/templates/registration/domain_request.html]: It looks like they are if you decorate the view [like this](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/settings/views.py#L93).

Comment (sravfeyn) [inline corehq/apps/registration/templates/registration/domain_request.html]: Haha, was thinking, may be this snippet is used too often and we should make a custom tag or something like that. Will add that.

Comment (esoergel) [inline corehq/apps/registration/templates/registration/domain_request.html]: :+1:

Comment (sravfeyn) [inline corehq/apps/registration/templates/registration/domain_request.html]: There are multiple views that use this form (DomainRegistrationForm) and adding this at multiple views that use one single template doesn't seem like the best way.  select2 is usually associated with an input field in a Form. So may be it's better to couple the select2 snippet with a Form than with a view?

Comment (esoergel) [inline corehq/apps/registration/templates/registration/domain_request.html]: Maybe, although that decorator requires access to the request object, which the form won't have.  You could probably come up wiht some clever way to insert it into forms automatically, but I think the best solution here is probably just to merge as-is.  If you agree, feel free to self-merge, I'm signed off.

**Comment by amsagoff** [comment]
FYI checked out a screenshot, looks awesome, no UAT needed

File: corehq/apps/registration/templates/registration/domain_request.html (hunk 1 above)
Comment (sravfeyn) [inline corehq/apps/registration/templates/registration/domain_request.html]: Used this (django feature)[https://docs.djangoproject.com/en/1.8/topics/forms/media/]

Comment (esoergel) [inline corehq/apps/registration/templates/registration/domain_request.html]: Oh nice, I'd forgotten that was a thing.

**Pull request merged**